pkg/operator/controller/canary/daemonset.go

```diff
@@ -73,3 +97,60 @@ func desiredCanaryDaemonSet(canaryImage string) *appsv1.DaemonSet {
 
 	return daemonset
 }
+
+// canaryDaemonSetChanged returns true if current and expected differ by the pod template's
+// node selector, tolerations, or container image reference.
+func canaryDaemonSetChanged(current, expected *appsv1.DaemonSet) (bool, *appsv1.DaemonSet) {
```

Comment:
Are namespace and command changes excluded?

Reply:
Not sure I follow. The canary namespace is reconciled separately in `namespace.go`.

For PR 561, you could add logic here to trigger updates when the canary container command changes.
Does that make sense?

Reply:
To determine if there was a change, you check node selector, tolerations, and container image reference.
I was just wondering if you should also be checking namespace and command changes here to determine if there was a change.

Reply:
Is there a situation we can think of where the canary daemonset's namespace would change? I think we want to keep that daemonset in the openshift-ingress-canary namespace always. 

For updates when the container command changes, nothing is changing `Spec.Template.Spec.Containers[0].Command` at the moment, so I dont think we need to check for updates there in this PR.

Reply:
Ok, yes, I'm getting ahead of myself, thinking about PR 561.